test_zipp.py

```diff
@@ -416,3 +417,29 @@ def test_inheritance(self, alpharep):
         for alpharep in self.zipfile_alpharep():
             file = cls(alpharep).joinpath('some dir').parent
             assert isinstance(file, cls)
+
+    def test_can_pickle_string_path(self):     
+        try:
+            path_1 = zipp.Path("/path/to/a/file.zip")
+            path_1_pickle = pickle.dumps(path_1)
+            path_1_load = pickle.loads(path_1_pickle)
+            assert path_1.root == path_1_load.root
+            path_2 = zipp.Path("/path/to/a/file.zip", at="something.txt")
+            path_2_pickle = pickle.dumps(path_2)
+            path_2_load = pickle.loads(path_2_pickle)
+            assert path_2.root == path_2_load.root
+        except TypeError as exec:
+            assert False, "TypeError: Path is not pickleable"
+
+    def test_can_pickle_pathlib_path(self):
+        try:
+            path_1 = zipp.Path(pathlib.Path("/path/to/a/file.zip"))
+            path_1_pickle = pickle.dumps(path_1)
+            path_1_load = pickle.loads(path_1_pickle)
+            assert path_1.root == path_1_load.root
+            path_2 = zipp.Path(pathlib.Path("/path/to/a/file.zip", at="something.txt"))
+            path_2_pickle = pickle.dumps(path_2)
+            path_2_load = pickle.loads(path_2_pickle)
+            assert path_2.root == path_2_load.root
+        except TypeError as exec:
+            assert False, "TypeError: Path is not pickleable"
```

Comment:
Why trap the error to replace it with another exception. Why not instead simply exercise the expected behavior and let exceptions produce failures?

```suggestion
        path_1 = zipp.Path("/path/to/a/file.zip")
        path_1_pickle = pickle.dumps(path_1)
        path_1_load = pickle.loads(path_1_pickle)
        assert path_1.root == path_1_load.root
        path_2 = zipp.Path("/path/to/a/file.zip", at="something.txt")
        path_2_pickle = pickle.dumps(path_2)
        path_2_load = pickle.loads(path_2_pickle)
        assert path_2.root == path_2_load.root

    def test_can_pickle_pathlib_path(self):
        path_1 = zipp.Path(pathlib.Path("/path/to/a/file.zip"))
        path_1_pickle = pickle.dumps(path_1)
        path_1_load = pickle.loads(path_1_pickle)
        assert path_1.root == path_1_load.root
        path_2 = zipp.Path(pathlib.Path("/path/to/a/file.zip", at="something.txt"))
        path_2_pickle = pickle.dumps(path_2)
        path_2_load = pickle.loads(path_2_pickle)
        assert path_2.root == path_2_load.root
```